### PR TITLE
Allow specific dynos to be restarted and viewing of dyno list and status

### PR DESCRIPTION
**Pull request by chaselee**
Allows for the following types of commands:

- `hubot heroku restart my-app` (restarts all dynos on app)
- `hubot heroku restart my-app worker` (restarts all worker dynos)
- `hubot heroku restart my-app web.2` (restarts only the web.2 dyno)

### CODE REVIEW

**Comment by chaselee** [comment]
@daemonsy let me know your thoughts!

**Comment by chaselee** [comment]
@daemonsy, added tests. should be good to go.

File: src/scripts/heroku-commands.coffee [hunk 1]
```diff
@@ -11,7 +11,7 @@
 #   hubot heroku info <app> - Returns useful information about the app
 #   hubot heroku releases <app> - Latest 10 releases
 #   hubot heroku rollback <app> <version> - Rollback to a release
-#   hubot heroku restart <app> - Restarts the app
+#   hubot heroku restart <app> <dyno> - Restarts the specified app or dyno/s
```
Comment (daemonsy) [inline src/scripts/heroku-commands.coffee]: Hey, thanks for this! It makes sense to do this plus some other commands
- Check what's the dyno formation for an app
- Scale web / worker dynos for an app

How about <app or dyno> or <app> or <dyno>  here? I haven't had a chance to look into the platform API yet, how do we specify the dyno to be reset? Is it a name or some id? I think it's nice to hint the format in the help message too.

**Comment by chaselee** [comment]
@daemonsy added another command (w/tests) to get the dyno list. Exactly mimics the Heroku cli `ps` command. Let me know what you think :smile:

**Comment by chaselee** [comment]
Example output for `hubot heroku dynos shield-global-watch`:

Damon: Dynos of shield-global-watch
=== web (1X): `forever server.js`
web.1: up 2015/01/01 07:00:00 (~ 5 months ago)
web.2: crashed 2015/01/01 07:00:00 (~ 5 months ago)

=== worker (2X): `celery worker`
worker.1: up 2015/06/01 08:00:00 (~ 3 days ago)

**Comment by chaselee** [comment]
any thoughts @daemonsy?

**Comment by daemonsy** [comment]
:+1: Thanks for this, it looks good to me. Merging it and releasing after doing a test internally :smile:

**Pull request merged**